src/code/sys_matrix.c

```diff
@@ -698,47 +702,70 @@ void Matrix_Transpose(MtxF* mf) {
     mf->yz = temp;
 }
 
-void func_800D1FD4(MtxF* mf) {
+/**
+ * Replaces the rotation in the matrix at the top of the stack, keeping scale and translation untouched.
```

Comment:
I think this function also combines the scales from `mf` with `sCurrentMatrix`?
(I don't think `mf`'s 3x3 section is normalized?)
This would match the final line of this comment, "*then the 3x3 part of `sCurrentMatrix` becomes `\f$ \texttt{mf} \cdot S \f$`*".

I might rewrite the first line to something like this? 
```suggestion
 * Modifies matrix at the top of the stack, by setting the rotation to zero, then applying the scale and rotation from `mf`. This keeps the translation untouched.
```
(I also think it helps to mention `mf` explicitly, to make it more clear that `sCurrentMatrix` is the thing being modified?)

(This is a conceptually kind of a weird operation, isn't it?)

Reply:
Yes `mf` can actually be anything, not necessarily a rotation or even a rotation*scale

I may have named this based on usage tbh, since it is exclusively used as `Matrix_ReplaceRotation(&globalCtx->billboardMtxF);`
billboarding is for making geometry face the camera, `billboardMtxF` is a rotation matrix which is the inverted view matrix

So a more accurate name would be `Matrix_Set3x3KeepScale` but it's also not a very good name is it

Reply:
I don't mind the function name -- this function is weird enough that any name will be weird.

I do think the comment should be updated though because it does not keep the scale untouched.